spring-cloud-commons/src/main/java/org/springframework/cloud/client/hedger/LatencyPercentileHedgingPolicy.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2013-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.client.hedger;
+
+import java.time.Duration;
+import java.util.concurrent.atomic.DoubleAdder;
+import java.util.concurrent.atomic.LongAdder;
+import java.util.function.Predicate;
+
+import io.micrometer.core.instrument.Clock;
+import io.micrometer.core.instrument.distribution.DistributionStatisticConfig;
+import io.micrometer.core.instrument.distribution.HistogramSnapshot;
+import io.micrometer.core.instrument.distribution.TimeWindowMax;
+import io.micrometer.core.instrument.distribution.TimeWindowPercentileHistogram;
+
+import org.springframework.lang.Nullable;
+import org.springframework.web.reactive.function.client.ClientRequest;
+import org.springframework.web.reactive.function.client.ClientResponse;
+
+/**
+ * @author Csaba Kos
+ * @author Kevin Binswanger
+ */
+public class LatencyPercentileHedgingPolicy implements HedgingPolicy {
+	private final Predicate<ClientRequest> shouldTrackRequestPredicate;
+	private final int numHedgedRequests;
```

Comment:
Should we provide an ability to report on, or answer what the current histogram is looking like at any point in time?